src/families/stellar/bridge/libcore.js

```diff
@@ -0,0 +1,261 @@
+
+    return signers.length > 1;
+  });
+
+const getTransactionStatus = async (a, t) => {
+  const errors = {};
+  const warnings = {};
+  const useAllAmount = !!t.useAllAmount;
+
+  if (a.freshAddress === t.recipient) {
+    errors.recipient = new InvalidAddressBecauseDestinationIsAlsoSource();
+  } else {
+    const { recipientError, recipientWarning } = await validateRecipient(
+      a.currency,
+      t.recipient
+    );
+
+    if (recipientError) {
+      errors.recipient = recipientError;
+    }
+
+    if (recipientWarning) {
+      warnings.recipient = recipientWarning;
+    }
+  }
+
+  if (await isAccountIsMultiSign(a)) {
+    errors.recipient = new StellarSourceHasMultiSign();
+  }
+
+  if (!t.fees || !t.baseReserve) {
+    errors.fees = new FeeNotLoaded();
+  }
+
+  let estimatedFees = !t.fees ? BigNumber(0) : t.fees;
+  let baseReserve = !t.baseReserve ? BigNumber(0) : t.baseReserve;
+
+  let amount = !useAllAmount
+    ? t.amount
+    : a.balance.minus(baseReserve).minus(estimatedFees);
+  let totalSpent = !useAllAmount
+    ? amount.plus(estimatedFees)
+    : a.balance.minus(baseReserve);
+
+  if (useAllAmount) {
+    warnings.amount = new StellarMinimumBalanceWarning();
+  }
+
+  if (
+    !errors.amount &&
+    amount
+      .plus(estimatedFees)
+      .plus(baseReserve)
+      .gt(a.balance)
+  ) {
+    errors.amount = new NotEnoughBalance();
+  }
+
+  if (
+    !errors.recipient &&
+    !errors.amount &&
+    (amount.lt(0) || totalSpent.gt(a.balance))
+  ) {
+    errors.amount = new NotEnoughBalance();
+    totalSpent = BigNumber(0);
+    amount = BigNumber(0);
+  }
+
+  if (!errors.amount && amount.eq(0)) {
+    errors.amount = new AmountRequired();
+  }
+
+  // if amount < 1.0 you can't
+  if (
+    !errors.amount &&
+    !(await checkRecipientExist(a, t.recipient)) &&
+    amount.lt(10000000)
+  ) {
+    errors.amount = new StellarNewAccountMinimumTransaction();
+  }
+
+  if (t.memoType && t.memoValue) {
+    try {
+      getMemoValidation(t.memoType, t.memoValue);
+    } catch (e) {
+      //We use transaction here to block screen on ledger-live-mobile
+      errors.transaction = e;
+    }
+  }
+
+  return Promise.resolve({
+    errors,
+    warnings,
+    estimatedFees,
+    amount,
+    totalSpent
+  });
+};
+
+const prepareTransaction = async (a, t) => {
+  let memoTypeRecommended = t.memoTypeRecommended;
+  let networkInfo = t.networkInfo;
+
+  if (!networkInfo) {
+    const ni = await getAccountNetworkInfo(a);
+    invariant(ni.family === "stellar", "stellar networkInfo expected");
+    networkInfo = ni;
+  }
+
+  const fees = t.fees || networkInfo.fees;
+  const baseReserve = t.baseReserve || networkInfo.baseReserve;
+  let memoType = t.memoType;
+
+  if (memoType === null) {
+    const { recipientError } = await validateRecipient(a.currency, t.recipient);
+    if (!recipientError) {
+      memoType = await memoTypeCheck(t.recipient);
+      memoTypeRecommended = memoType !== null ? true : false;
+    }
+  }
```

Comment:
Tell me if it's clearer without mutation (it's better to unit test this behaviour later if you have already a simple function):
```
const getMemoData = () => {
  if (t.memoType) {
    return { memoType: t.memoType, memoTypeRecommended: true }
  } else {
    const { recipientError } = await validateRecipient(a.currency, t.recipient);
    if (!recipientError) {
      const memoType = await memoTypeCheck(t.recipient);
      const memoTypeRecommended = memoType !== null ? true : false;
      return { memoType, memoTypeRecommended }
    } else {
      return { memoType: undefined, memoTypeRecommended: false }
    }
  }
}

const { memoType, memoTypeRecommended } = getMemoData();
```